runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkState.java

```diff
@@ -124,28 +122,30 @@ private static String elapsedString(Instant start, Instant end) {
    */
   synchronized ActivateWorkResult activateWorkForKey(ExecutableWork executableWork) {
     ShardedKey shardedKey = executableWork.work().getShardedKey();
-    Deque<ExecutableWork> workQueue = activeWork.getOrDefault(shardedKey, new ArrayDeque<>());
+    long shardingKey = shardedKey.shardingKey();
+    LinkedHashMap<WorkId, ExecutableWork> workQueue =
+        activeWork.getOrDefault(shardingKey, new LinkedHashMap<>());
```

Comment:
can we use activeWork.computeIfAbsent?
Then we can avoid the put in the common case there is nothing queued.

Reply:
done.